mono_repo/test/yaml_write_test.dart

```diff
@@ -0,0 +1,695 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:mono_repo/src/yaml.dart';
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart' as y;
+
+import 'shared.dart';
+
+void _asciiTest(List<int> bytes) {
+  test(bytes.toString(), () {
+    var source = ascii.decode(bytes);
+    var output = toYaml(source);
+    var yaml = y.loadYaml(output);
+    expect(yaml, source, reason: bytes.toString());
+  });
+}
+
+void main() {
+  group('ascii', () {
+    for (var i = 0; i < 128; i++) {
+      _asciiTest([i]);
+    }
+  });
+
+  group('naughty strings', () {
+    for (var source in _stringEscapeSamples.followedBy(_naughtyStrings)) {
+      _testRoundTrip(source);
+    }
+  });
+
+  for (var item in [
```

Comment:
lets move this list into a top level variable or something, its pretty hard to find the actual block for the loop, especially with the listed maps :joy_cat: